tests/Fixer/Casing/MagicMethodCasingFixerTest.php

```diff
@@ -0,0 +1,304 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Casing;
+
+use PhpCsFixer\Fixer\Casing\MagicMethodCasingFixer;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Casing\MagicMethodCasingFixer
+ */
+final class MagicMethodCasingFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string $expected
+     * @param string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        $fixerReflection = new \ReflectionClass(MagicMethodCasingFixer::class);
+        $property = $fixerReflection->getProperty('magicNames');
+        $property->setAccessible(true);
+        $allMethodNames = $property->getValue();
```

Comment:
I think we should avoid relying on such internal properties to generate test cases: if one removes values from it, tests won't fail as they won't run at all.

Reply:
Exactly. When removing code, if tests don't fail, one might think the removed code was not used or some other change replaced it, which is not true. A missing test when adding code is easily spotted during review.

Reply:
In both cases this will have to be spotted during review. But it is different when nobody spots it. When removing code *and* test, you probably break something for users. When adding new code without test, you just add something that (probably) works as expected, or at least doesn't break existing tests.

In general, tests should be as much decoupled from the code itself as possible. The point is to be defensive. This is similar to [Symfony doc's advice regarding hardcoded URLs in tests](https://symfony.com/doc/current/best_practices/tests.html#hardcode-urls-in-a-functional-test).

Reply:
Well this is one pattern against another; decoupled test or DRY: maintaining the same list of static values at two spots in the code.
I understand the arguments; I just out weight the value of not having two list + generating tests cases so I don't forget one, over the more theoretical decoupling of test argument.